### PR TITLE
Update rust-selectors

### PR DESCRIPTION
**Pull request by untitaker**

### CODE REVIEW

File: src/select.rs [hunk 1]
```diff
@@ -6,13 +6,19 @@ use std::ascii::AsciiExt;
...
+///
+/// Copied from rust-selectors.
+static SELECTOR_WHITESPACE: &'static [char] = &[' ', '\t', '\n', '\r', '\x0C'];
+
+#[derive(Debug)]
 pub struct KuchikiSelectors;
 
 impl SelectorImpl for KuchikiSelectors {
     type NonTSPseudoClass = PseudoClass;
     type PseudoElement = PseudoElement;
```
Comment (untitaker) [inline src/select.rs]: There are some typedefs missing here ([see selectors source](https://github.com/servo/rust-selectors/blob/cd0230444b1df96bfbdce509c0fab9f7f737c45d/src/parser.rs#L44)), but I couldn't figure out what those were good for, or what the entire trait was good for.

Comment (SimonSapin) [inline src/select.rs]: Sorry for the lack of documentation in the selectors crate. It became more generic over the string types used for various bits of a selector. Based on types used in kuchiki::tree, I suggest:

``` rust
    type AttrValue = String;
    type Identifier = String;
    type ClassName = String;
    type LocalName = Atom;
    type NamespaceUrl = Namespace;
    type NamespacePrefix: String;
    type BorrowedNamespaceUrl: Namespace;
    type BorrowedLocalName: Atom;
```

where `Atom` and `Namespace` are from the string-cache crate.

Comment (SimonSapin) [inline src/select.rs]: We could take this opportunity to simplify Kuchiki and use `String` everywhere, but string-cache makes comparison faster and selector matching makes a lot of string comparison. This was a very significant performance improvement in Servo, but I don’t know if it’s as critical in Kuchiki.

Comment (untitaker) [inline src/select.rs]: I don't really understand why selector doesn't come with some implementation of SelectorImpl.

Comment (untitaker) [inline src/select.rs]: Updated and rearranged typedefs such that they match the `rust-selector` sourcecode.

**Comment by untitaker** [comment]
I changed two of your suggestions to stay compatible with the types previously used (both: `String => Atom`).

Also BTW it appears that you changed `Namespace` to `NamespaceUrl` in master.

**Comment by untitaker** [comment]
@SimonSapin This now passes all tests.

**Comment by SimonSapin** [comment]
> I don't really understand why selector doesn't come with some implementation of SelectorImpl.

The point of `SelectorImpl` is to allow users of `selectors` to specify things they might want to do differently. But yeah, I suppose there could be a "reasonable default" implementation.

> Also BTW it appears that you changed Namespace to NamespaceUrl in master.

Yes, I made another change that turned out to be breaking: recording the namespace prefixes to support serialization. This rename is to disambiguate with the new `NamespacePrefix` associated type. This is all in version 0.10 which I had neglected to publish on crates.io. I’ve just done so.

(There is no hurry to update to that, old versions stay around.)

---

Looks good! Please squash into one commit and this is good to go.

**Comment by untitaker** [comment]
You can do this with the GitHub UI, otherwise I will squash tomorrow

**Comment by SimonSapin** [comment]
Oh that’s right, I forgot about that feature.

**Comment by SimonSapin** [comment]
Thanks!

**Comment by untitaker** [comment]
Thanks to you as well. Could you also release a new version?

On 17 August 2016 01:18:41 CEST, Simon Sapin notifications@github.com wrote:

> Thanks!
> 
> ## 
> 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/SimonSapin/kuchiki/pull/27#issuecomment-240268919

## 

Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by SimonSapin** [comment]
v0.3.3 is up.

**Comment by untitaker** [comment]
Thanks!

On 17 August 2016 01:27:12 CEST, Simon Sapin notifications@github.com wrote:

> v0.3.3 is up.
> 
> ## 
> 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/SimonSapin/kuchiki/pull/27#issuecomment-240270339

## 

Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Pull request merged**